### ISSUE TITLE
Subcommands that share a name with a parent will not autocomplete

### ISSUE THREAD

**Issue by mitchellh** [posener/complete#46]
The title isn't great. I'll explain. First, thanks for a great library. 🎉 

I have a diff at the bottom of this issue with a fix, but I'm not sure how you want to unit test it so I didn't open a PR. Its a one-line fix.

## The Problem

We have a CLI with the following two commands:

* `nomad job status`
* `nomad status`

They each implement their own arg autocompletion func. However, we noticed that only the outer one was ever being called. i.e. when you did `nomad job status <tab>`, it was returning the same output for `nomad status <tab>`. Further confusing: it seemed to call both autocomplete funcs, but only outputted the outer one. 

We expected the behavior to only call the inner one.

A failing structure is shown below (where each Args PredictFucn outputs a different thing):

```go
complete.Command{
    Sub: {
        "job": {
            Sub: {
                "status": {
                    Sub:         {},
                    Flags:       {},
                    GlobalFlags: {},
                    Args:        complete.PredictFunc {...},
                },
            },
            Flags:       {},
            GlobalFlags: {},
            Args:        nil,
        },
        "status": {
            Sub:         {},
            Flags:       {},
            GlobalFlags: {},
            Args:        complete.PredictFunc {...},
        },
    },
    Flags:       {},
    GlobalFlags: {},
    Args:        nil,
}
```

## The Solution

I found the solution is to set `only = true` in `c.predict `when there is a subcommand found. This properly short-circuits the search. Otherwise, a parent matching command is found and overwrites any results. We also don't want the results merged, since the parent command does something totally different. 

Here is the diff:

```diff
diff --git a/command.go b/command.go
index eeeb9e0..42b7f43 100644
--- a/command.go
+++ b/command.go
@@ -87,6 +87,7 @@ func (c *Command) predict(a Args) (options []string, only bool) {
        // if a sub command was entered, we won't add the parent command
        // completions and we return here.
        if subCommandFound {
+         only = true
                return
        }
```

We'd love to see this fixed!

Again, thanks for a great library. We're rolling this out to all HashiCorp tooling and its been great so far. This is the first bug we've ran into.

**Comment by posener** [comment]
Hi @mitchellh 
Thanks for the nice words, this detailed issue, and the proposed change.
I agree that this is a bug that should be solved :-)
I am afraid that the fix is not that simple, as I can see on problem for this change (I think): 
Suppose the subcommand returns `only == true`, then the parent's `GlobalFlags` won't be added to the completion options, because of this [`return` statement](https://github.com/posener/complete/blob/master/command.go#L74). `GlobalFlags` are flags that should appear even if a subcommand completion was found.

**Comment by mitchellh** [comment]
Ah hah. As I suspected, its not so simple. :) Thanks for the clarity.

We'd love to see this fixed! Please let me know if I can help in any way.

**Comment by posener** [comment]
PR is always welcome :-)

**Comment by mitchellh** [comment]
This popped up in another way, too:

Any commands with 2 more levels of nested subcommands will not autocomplete at all. Example:

* `nomad operator raft <tab>` will not ever work
* `nomad operator <tab>` will work

Verified the same fix fixes this too, caveats of what you said understood.

**Comment by posener** [comment]
Fixed with #48
